snuba/query/snql/parser.py

```diff
@@ -136,7 +141,7 @@
     low_pri_tuple         = low_pri_op space* high_pri_arithmetic
     high_pri_tuple        = high_pri_op space* arithmetic_term
 
-    arithmetic_term       = space* (function_call / numeric_literal / subscriptable / column_name / parenthesized_arithm)
+    arithmetic_term       = space* (function_call / subscriptable / simple_term / parenthesized_arithm)
```

Comment:
NULL should not be a valid term here. Arithmetic terms are used in conjunction with `low_pri_op` or `hi_pri_op` which are `+ - / * ` and are not supposed to have NULL as an operand.
Also quoted_literal should not be valid here. Any reason to add it ?

Reply:
This is also ultimately used as in the SELECT column as well (selected_expression -> low_pri_arithmetic -> high_pri_arithmetic -> arithmetic_term). The way this is written, it's hard to separate out the different use cases the way you're suggesting. That's why I added it here. I'll add a tech debt story to rearchitect these regexes to allow us to be more specific when validating.

Reply:
Also, it is valid to do math with NULL. It just always results in a result of NULL.